### PR TITLE
Errno review

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Goes some way to address #7845
Most obvious cases fixed, although there remain some. Also, consequently uses stringerror() instead of variations of string(strerror(errno))

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/nameserver.cc [hunk 1]
```diff
@@ -103,8 +103,8 @@ void UDPNameserver::bindIPv4()
     s=socket(AF_INET,SOCK_DGRAM,0);
 
     if(s<0) {
-      g_log<<Logger::Error<<"Unable to acquire UDP socket: "+string(strerror(errno)) << endl;
-      throw PDNSException("Unable to acquire a UDP socket: "+string(strerror(errno)));
+      g_log<<Logger::Error<<"Unable to acquire UDP socket: "+stringerror() << endl;
```
Comment (chbruyand) [inline pdns/nameserver.cc]: Technically `strerror()` can override `errno` no ?

File: pdns/syncres.cc [hunk 2]
```diff
@@ -3108,7 +3108,8 @@ bool SyncRes::doResolveAtThisIP(const std::string& prefix, const DNSName& qname,
       /* -1 means server unreachable */
       s_unreachables++;
       d_unreachables++;
-      LOG(prefix<<qname<<": error resolving from "<<remoteIP.toString()<< (doTCP ? " over TCP" : "") <<", possible error: "<<strerror(errno)<< endl);
+      // XXX questionable use of errno
```
Comment (chbruyand) [inline pdns/syncres.cc]: indeed

File: pdns/tcpreceiver.cc [hunk 3]
```diff
@@ -1291,15 +1292,16 @@ TCPNameserver::TCPNameserver()
...
     if(bind(s, (const sockaddr*)&local, local.getSocklen())<0) {
+      int err = errno;
       close(s);
-      if( errno == EADDRNOTAVAIL && ! ::arg().mustDo("local-ipv6-nonexist-fail") ) {
+      if( err == EADDRNOTAVAIL && ! ::arg().mustDo("local-ipv6-nonexist-fail") ) {
         g_log<<Logger::Error<<"IPv6 Address " << *laddr << " does not exist on this server - skipping TCP bind" << endl;
         continue;
       } else {
-        g_log<<Logger::Error<<"Unable to bind to TCPv6 socket" << *laddr << ": "<<strerror(errno)<<endl;
+        g_log<<Logger::Error<<"Unable to bind to TCPv6 socket" << *laddr << ": "<<strerror(err)<<endl;
```
Comment (chbruyand) [inline pdns/tcpreceiver.cc]:
```suggestion
        g_log<<Logger::Error<<"Unable to bind to TCPv6 socket" << *laddr << ": "<<stringerror(err)<<endl;
```

**Comment by omoerbeek** [comment]
Fixed mentioned cases.

**Pull request merged**